File: src/main/java/com/event/Flowvent/entity/Event.java
package com.event.Flowvent.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;
import java.util.List;

@Entity
@Data
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String description;
    private LocalDate date;
    private Integer maximumCapacity;
    private Double ticketPrice;

    @OneToMany(mappedBy = "event", cascade = CascadeType.ALL)
    private List<Ticket> tickets;
}

File: frontend/src/types/event.ts
export interface Event {
  id: number
  title: string
  description: string
  date: string
  maximumCapacity: number
  ticketPrice: number
  soldTickets: number
  availableTickets: number
}

export interface EventCreateRequest {
  title: string
  description: string
  date: string
  maximumCapacity: number
  ticketPrice: number
}

export type EventUpdateRequest = EventCreateRequest